### PR TITLE
Display possible values when using in() with array

### LINKED ISSUE #385: Display possible values when using in() with array
The exception message for `in` with an array of strings will display `Array` instead of the haystack values.

`in('a', 'b')->validate('x')`
currently results in
`"x" must be in (Array)`
but should result in
`"x" must be in ('foo', 'bar')`

### PR DESCRIPTION
**Pull request by nochso**
Re-submitting PR #386 based off master
Fixes #385

### CODE REVIEW

**Comment by nochso** [comment]
```
$array = [    
    1,
    "respect",
    [1,2 , [ "respect" ]]
]
```
`(1, 'respect', Array)`
or
`(1, 'respect', (1, 2, ('respect')))` with a maximum depth, indicated with `, ... `

Keep in mind that stringify() itself does not add the surround `()` so this might have to be reworked.

File: library/Exceptions/ValidationException.php [hunk 1]
```diff
@@ -27,6 +27,10 @@ class ValidationException extends InvalidArgumentException implements Validation
             self::STANDARD => 'Data validation failed for %s',
         ),
     );
+    /**
+     * @var int Maximum depth when stringifying nested arrays
+     */
+    public static $maxDepthStringify = 2;
```
Comment (henriquemoody) [inline library/Exceptions/ValidationException.php]: Make it private, and increase that to 3

File: library/Exceptions/ValidationException.php [hunk 2]
```diff
@@ -57,6 +61,31 @@ public static function stringify($value)
         }
     }
 
+    /**
+     * @param $value
+     * @return string
+     */
+    public static function stringifyArray($value, $depth = 0)
```
Comment (henriquemoody) [inline library/Exceptions/ValidationException.php]: Make it private and also fix the wrong docblock.

File: library/Exceptions/ValidationException.php [hunk 3]
```diff
@@ -57,6 +61,31 @@ public static function stringify($value)
         }
     }
 
+    /**
+     * @param $value
+     * @return string
```
Comment (henriquemoody) [inline library/Exceptions/ValidationException.php]: New line before `@return`

File: library/Exceptions/ValidationException.php [hunk 4]
```diff
@@ -57,6 +61,31 @@ public static function stringify($value)
         }
     }
 
+    /**
+     * @param $value
```
Comment (henriquemoody) [inline library/Exceptions/ValidationException.php]: Missing type and missing `$depth`

**Comment by henriquemoody** [comment]
After the changes you also need to squash all commits at once.

**Comment by nochso** [comment]
Ok, I've made the changes and squashed it into a single commit.

**Comment by henriquemoody** [comment]
I did it manually on commit 2de6e013, because I also fixed some coding standards in your commit (to not bother you).

Thank you for contributing! :panda_face:

**Comment by nochso** [comment]
Gladly. I just learned you could rebase / squash like that even though I already pushed on that branch, so thank you too :)

**Pull request closed without merging**